aspnetcore/fundamentals/portable-object-localization.md

```diff
@@ -0,0 +1,218 @@
+<p>@Localizer.Plural(1, "There is one item.", "There are {0} items.")</p>
+<p>@Localizer.Plural(2, "There is one item.", "There are {0} items.")</p>
+<p>@Localizer.Plural(5, "There is one item.", "There are {0} items.")</p>
+```
+
+**Note:** In a real world scenario, a variable would be used to represent the count. Here, we repeat the same code with three different values to expose a very specific case.
+
+Upon switching cultures, you see the following:
+
+For `/Home/About`:
+
+```html
+There is one item.
+There are 2 items.
+There are 5 items.
+```
+
+For `/Home/About?culture=fr`:
+
+```html
+Il y a un élément.
+Il y a 2 éléments.
+Il y a 5 éléments.
+```
+
+For `/Home/About?culture=cs`:
+
+```html
+Existuje jedna položka.
+Existují 2 položky.
+Existuje 5 položek.
+```
+
+Note that for the Czech culture, the three translations are different. The French and English cultures share the same construction for the two last translated strings.
+
+## Advanced tasks
+
+### Contextualizing strings
+
+Applications often contain the same strings to be translated in several places, while requiring the flexibility to define different translations. A PO file supports the notion of a *context*, which can be used to categorize the string being represented.
+
+The PO localization services can use the name of the full class or the view that is used when translating a string. This is accomplished by setting the value on the `msgctx` entry.
+
+Considering the previous example, the entry could have been written as:
+
+*fr.po*
+
+```text
+msgctx Views.Home.About
+msgid "Hello world!"
+msgstr "Bonjour le monde!"
+```
+
+When no context is specified, it's used as a fallback for any string whose context doesn't match any specific one.
+
+### Changing the location of PO files
+
+You can change the default location of PO files in the application by setting a base folder name. This is accomplished with the following code in the `ConfigureServices` method of *Startup.cs*:
+
+```csharp
+services.AddPortableObjectLocalization(options => options.ResourcesPath = "Localization");
+```
+
+In this example, the PO files are loaded from the *Localization* folder.
+
+### Implementing a custom logic for finding localization files
+
+When more complex logic is needed to locate PO files, the `OrchardCore.Localization.PortableObject.ILocalizationFileLocationProvider` interface can be implemented
+and registered as a service. This is useful when PO files can be stored in varying locations or when the files have to be found within a hierarchy of folders.
+
+### Using a different default pluralized language
+
+The package includes a `Plural` extension method that is specific to two plural forms, as this is the most common. If your main language is different and requires more plural forms, for instance, create your own extension method. This way, you won't need to provide any localization file for the default language &mdash; the original strings are already available directly in the code.
```

Comment:
The package includes a Plural extension method that is specific to two plural forms. For languages that require more plural forms, create an extension method. With a extension method, you won't need to provide any localization file for the default language — the original strings are already available directly in the code.
You can use the more generic Plural(int count, string[] pluralForms, params object[] arguments) overload which accepts a string array of translations.